woocommerce/woocommerce-gateway-stripe
# Remove hard requirement for customer phone numbers with Apple Pay

Fixes #1303.

The entirety of the fix is contained in https://github.com/woocommerce/woocommerce-gateway-stripe/commit/3654b4e11843711fa241a90c8476feac9336484f. The other 2 commits are some changes to the linting config, and addressing some pre-existing linting issues. I know it's not strictly part of this issue, but I was touching the code anyway, so I thought I'd clear up some of those.

~~@bmccotter @allendav Do you think the language describing the new option in the Stripe settings is good enough, or would you like to see some changes there?~~

~~Also, should this even be a new option in the settings? Maybe it would be better to add this as a filter instead of a new option in the plugin settings? And maybe we should just respect the WC core setting from the theme customizer?~~

The implementation now respects the [Checkout settings](https://docs.woocommerce.com/document/woocommerce-customizer/#section-17) in the customizer. The fix is now in 3654b4e11843711fa241a90c8476feac9336484f (like before) and 6507aacd1268612241bd5d8095710e04df3b28bf (one-line change).

How do we make this as painless as possible?

#### Changes proposed in this Pull Request:

- Adds a checkbox in the Stripe settings giving merchants the option to require a phone number when customers checkout with Apple Pay.
- Phone numbers are **not required** by default.

#### Testing instructions

1. Delete the `woocommerce_checkout_phone_field` row in the `wp_options` DB table.
1. Try starting a payment using Apple Pay.
2. Confirm that you **need** to add a phone number to the contact details.
3. Now open the WP theme customizer, and navigate to `WooCommerce > Checkout`.
4. Set the `Require Phone Number` option to `optional`.
5. Try starting a payment using Apple Pay.
6. Confirm that you **don't need** to add a phone number to the contact details.
7. Now set the `Require Phone Number` option back to `required`.
8. Try starting a payment using Apple Pay.
9. Confirm that you **need** to add a phone number to the contact details.

The reason we switch from `required` to `optional` and back to `required` is because in step 3 we're relying on the default value of `get_option`. In steps 7 and 10 we're checking the actual value of the setting retreived by `get_option`.

-------------------
- [x] Did you make changes, or create a **new .js file**? If **Gruntfile.js** exists in the repo, make sure to run `grunt`. _I ran `npm run uglify`, I think that's enough?_.
